cmd/crossplane/core/core.go

```diff
@@ -147,6 +146,10 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		//nolint:revive // This is long and easier to read with punctuation.
 		return errors.New("Crossplane removed support for external secret stores. The --enable-external-secret-stores flag will be removed in a future release.")
 	}
+	if c.Registry != "" {
+		//nolint:revive // This is long and easier to read with punctuation.
+		return errors.New("Crossplane removed support for the --registry flag, as a default registry is not supported. Please fully qualify images by explicitly providing a registry.")
```

Comment:
```suggestion
		return errors.New("The --registry flag is no longer supported since support for a default registry value has been removed. Please ensure that all packages have fully qualified names that explicitly state their registry. This also applies to all of a packages dependencies.")
```